model.py

```diff
@@ -231,8 +234,11 @@ def Train(self,
             use_3_trees=self.use_3_trees,
             cat_idx=self.cat_idx,
             num_idx=self.num_idx)
+        # in multi-class classification, the target has to be binary
+        # as each tree is a per-class regressor
         tree.Fit(X_tree,
-                 (y_ensemble == kth_tree).astype(np.float64) if self.loss_.is_multi_class else y_ensemble,
+                 (y_ensemble == kth_tree).astype(np.float64) if self.loss_.is_multi_class
```

Comment:
Maybe (y_ensemble == kth_tree).astype(np.float64) if self.loss_.is_multi_class else y_ensemble can be its own statement a few lines before rather than within the function call for more readability?